src/main/ruby/truffleruby/core/truffle/range_operations.rb

```diff
@@ -110,7 +109,11 @@ def self.range_cover?(range, other)
     def self.cover?(range, value)
       # MRI uses <=> to compare, so must we.
       beg_compare = (range.begin <=> value)
-      return false unless beg_compare
+
+      unless beg_compare
+        return false if range.begin # not beginless
+        beg_compare = -1
+      end
```

Comment:
Same as `cover_range?`, hard to follow because of the existing code.

Proposal (completely untested):
```ruby
def self.cover?(range, value)
  # Check lower bound.
  if !Primitive.nil?(range.begin)
    # MRI uses <=> to compare, so must we.
    cmp = (range.begin <=> value)
    return false if Compare.compare_int(cmp) > 0
  end

  # Check upper bound.
  if !Primitive.nil(range.end)
    value = value + (range.exclude_end? 1 : 0)
    cmp = (value <=> range.end)
    return false if Comparable.compare_int(cmp) > 0
  end

  true
end
```

Reply:
Also much more beautiful, thanks!

Also, `r_less` is no longer being used, so I commented it out; not sure if it should be removed altogether.

Reply:
Yes, let's remove `range_less` if it's unused :)
(we can always use the git history to restore it if it's useful in the future)